cmd/teleporter-cli/transaction.go

```diff
@@ -30,46 +33,105 @@ var transactionCmd = &cobra.Command{
 for Teleporter and Warp log events. When corresponding log events are found,
-the command parses to log event fields to a more human readable format.`,
+the command parses to log event fields to a more human readable format. Optionally pass -d 
+or --debug for extra transaction output. This may require debug enpoints enable on your RPC node`,
 	Args: cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
-		receipt, err := client.TransactionReceipt(context.Background(),
-			common.HexToHash(args[0]))
-		cobra.CheckErr(err)
-
-		for _, log := range receipt.Logs {
-			if log.Address == teleporterAddress {
-				logger.Info("Processing Teleporter log", zap.Any("log", log))
-
-				event, err := teleporterABI.EventByID(log.Topics[0])
-				cobra.CheckErr(err)
-
-				out, err := teleportermessenger.FilterTeleporterEvents(log.Topics, log.Data, event.Name)
-				cobra.CheckErr(err)
-				logger.Info("Parsed Teleporter event", zap.String("name", event.Name), zap.Any("event", out))
-			}
-
-			if log.Address == common.HexToAddress(warpPrecompileAddress) {
-				logger.Debug("Processing Warp log", zap.Any("log", log))
-
-				unsignedMsg, err := warp.UnpackSendWarpEventDataToMessage(log.Data)
-				cobra.CheckErr(err)
-
-				warpPayload, err := warpPayload.ParseAddressedCall(unsignedMsg.Payload)
-				cobra.CheckErr(err)
-
-				teleporterMessage, err := teleportermessenger.UnpackTeleporterMessage(warpPayload.Payload)
-				cobra.CheckErr(err)
-				logger.Info("Parsed Teleporter message",
-					zap.String("warpMessageID", unsignedMsg.ID().Hex()),
-					zap.String("teleporterMessageNonce", teleporterMessage.MessageNonce.String()),
-					zap.Any("message", teleporterMessage))
-			}
+		txHash := common.HexToHash(args[0])
+		if debug {
+			printTransaction(cmd, txHash)
+			traceTransaction(cmd, txHash)
 		}
+		checkReceipt(cmd, txHash)
 		cmd.Println("Transaction command ran successfully")
 	},
 }
 
+func checkReceipt(cmd *cobra.Command, txHash common.Hash) {
+	receipt, err := client.TransactionReceipt(context.Background(), txHash)
+	cobra.CheckErr(err)
+
+	for _, log := range receipt.Logs {
+		printTeleporterLogs(cmd, log)
+		printWarpLogs(cmd, log)
+	}
+}
+
+func printTeleporterLogs(cmd *cobra.Command, log *types.Log) {
+	if log.Address != teleporterAddress {
+		return
+	}
+
+	logJson, err := json.MarshalIndent(log, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println("Teleporter Log:\n" + string(logJson) + "\n")
+
+	event, err := teleporterABI.EventByID(log.Topics[0])
+	cobra.CheckErr(err)
+
+	out, err := teleportermessenger.FilterTeleporterEvents(log.Topics, log.Data, event.Name)
+	cobra.CheckErr(err)
+
+	outJson, err := json.MarshalIndent(out, "", "  ")
+	cobra.CheckErr(err)
+
+	cmd.Println(event.Name + " Log:")
+	cmd.Println(string(outJson) + "\n")
+}
+
+func printWarpLogs(cmd *cobra.Command, log *types.Log) {
+	if log.Address != common.HexToAddress(warpPrecompileAddress) {
+		return
```

Comment:
An error message here would be helpful.

Reply:
I restructured this a bit to make more sense.